src/status_im/common/standard_authentication/enter_password/view.cljs

```diff
@@ -2,51 +2,61 @@
     [status-im.common.standard-authentication.password-input.view :as password-input]
     [status-im.contexts.profile.utils :as profile.utils]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
 (defn view
-  [{:keys [on-enter-password on-press-biometrics button-label button-icon-left]}]
-  (let [{:keys [key-uid customization-color] :as profile} (rf/sub [:profile/profile-with-image])
-        {:keys [error processing password]}               (rf/sub [:profile/login])
-        sign-in-enabled?                                  (rf/sub [:sign-in-enabled?])]
-    [:<>
-     [rn/view {:style style/enter-password-container}
-      [rn/view
-       [quo/text
-        {:accessibility-label :sync-code-generated
-         :weight              :bold
-         :size                :heading-1
-         :style               {:margin-bottom 4}}
-        (i18n/label :t/enter-password)]
-       [rn/view
-        {:style style/context-tag}
-        [quo/context-tag
-         {:type                :default
-          :blur?               true
-          :profile-picture     (profile.utils/photo profile)
-          :full-name           (profile.utils/displayed-name profile)
-          :customization-color customization-color
-          :size                24}]]
-       [password-input/view
-        {:on-press-biometrics on-press-biometrics
-         :processing          processing
-         :error               error
-         :default-password    password
-         :sign-in-enabled?    sign-in-enabled?}]
-       [quo/button
-        {:size                40
-         :container-style     style/enter-password-button
-         :type                :primary
-         :customization-color (or customization-color :primary)
-         :accessibility-label :login-button
-         :icon-left           button-icon-left
-         :disabled?           (or (not sign-in-enabled?) processing)
-         :on-press            (fn []
-                                (rf/dispatch [:set-in [:profile/login :key-uid] key-uid])
-                                (rf/dispatch [:profile.login/verify-database-password password
-                                              #(on-enter-password password)]))}
-        button-label]]]]))
+  [_]
+  (let [password-state (reagent/atom nil)]
+    (fn [{:keys [on-enter-password on-press-biometrics button-label button-icon-left]}]
+      (let [{:keys [key-uid customization-color]
+             :as   profile}                     (rf/sub [:profile/profile-with-image])
+            {:keys [error processing password]} (rf/sub [:profile/login])
+            sign-in-enabled?                    (rf/sub [:sign-in-enabled?])
+            on-change-password                  (fn [value]
+                                                  (when (not= value @password-state)
+                                                    (reset! password-state value)
+                                                    (rf/dispatch [:standard-auth/update-password
```

Comment:
Since we are talking more about performance these days, I'd like to mention a few things for this `enter-password` component I deem significant and that appear in many of our PRs. In no order of importance:

1. There's an unnecessary overhead to involve re-frame on every key press. Why are we committing to the app-db all the time?
2. Because we dispatch `:standard-auth/update-password` on every key press, this forces the parent `view` component to be rendered twice, for every key press. This problem cascades to its children `password-input/view` and `quo/button`. `input/view` itself is quite heavy. We can save a bunch of CPU cycles if we process the parent `view` component and its children only once on every key press.
3. The props passed to the `button` component are different on every Reagent pass because the `on-press` function is recreated over and over. This forces the button component to be re-processed by Reagent on every pass of the parent component.

Things that would allow us to render all components in `enter-password` once on every key press:

- Use local state and don't update the app-db at paths `[:profile/login :password]` and `[:profile/login :error]` on every key press. Use local state and then the whole `enter-password` component and its children would render once. The global state should be left for things that change less often.
- The profile `key-uid` doesn't change when the user is typing, so it can be moved outside the renderer, and then we can make the `on-press` of the button be created only once. This would allow Reagent to skip processing the button component entirely on every key press. The button component is not the cheapest too if we look at its implementation.

cc @flexsurfer A slightly different example than what we were discussing today in the perf channel. Still, a good example where multiple Reagent passes cascade down the component tree (and I haven't checked the other subcomponents, but on the surface they also have low-hanging fruits to optimize).